### PR TITLE
Hook Defiance up to threat

### PR DESCRIPTION
**Pull request by jokal2**

### CODE REVIEW

**Comment by jokal2** [comment]
This is fixed up and should be good to land now.

**Comment by jokal2** [comment]
@timhul  This is ready for review

File: Class/Warrior/TalentTrees/Protection.cpp [hunk 1]
```diff
@@ -126,3 +125,11 @@ void Protection::add_improved_sunder_armor(QMap<QString, Talent*>& talent_tier)
     add_talent_to_tier(talent_tier, talent);
 }
 
+void Protection::add_defiance(QMap<QString, Talent*>& talent_tier) {
+    auto talent = get_new_talent(warrior, "Defiance", "3RR", base_url + "ability/Ability_warrior_innerrage.png", 5,
+                                 "Increase the threat generated by your attacks by %1% while in Defensive Stance.",
+                                 QVector<QPair<unsigned, unsigned>> {{1, 1}}, QVector<SpellRankGroup*> {},
```
Comment (yellowmia11) [inline Class/Warrior/TalentTrees/Protection.cpp:131]: > {{1, 1}}
should be
{{3, 3}}

File: Class/Warrior/Buffs/DefensiveStanceBuff.cpp [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
 DefensiveStanceBuff::DefensiveStanceBuff(Warrior* warrior) :
-    SelfBuff(warrior, "Defensive Stance", NO_ICON, BuffDuration::PERMANENT, 1), warrior(warrior) {
+    SelfBuff(warrior, "Defensive Stance", NO_ICON, BuffDuration::PERMANENT, 1),
+    warrior(warrior),
+    TalentRequirer(QVector<TalentRequirerInfo*> {new TalentRequirerInfo("Defiance", 5, DisabledAtZero::No)}),
+    current_threat_mod(30) {
     this->hidden = true;
 }
 
+static std::vector<double> defiance_values {1.00, 1.03, 1.0609, 1.0927, 1.1255, 1.1592};
```
Comment (yellowmia11) [inline Class/Warrior/Buffs/DefensiveStanceBuff.cpp:16]: You can delete this line.

**Pull request merged**